File: scratch/test_assignments_platform.js
const fetch = require('node-fetch');

async function testPlatformAssignments() {
  const adminTokenResult = await fetch('http://localhost:3000/api/auth/login', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ email: 'admin@boozathink.com', password: 'admin' })
  });
  const adminData = await adminTokenResult.json();
  const token = adminData.token;

  if (!token) {
    console.error('Login failed.');
    return;
  }
  console.log('1. Logged in as Admin');

  // Test Profile
  const me = await fetch('http://localhost:3000/api/church/profile', {
    headers: { 'Authorization': `Bearer ${token}` }
  });
  const meData = await me.json();
  console.log('2. Profile:', meData);

  // Test Assignments
  const assigns = await fetch('http://localhost:3000/api/church/assignments/me', {
    headers: { 'Authorization': `Bearer ${token}` }
  });
  const assignsData = await assigns.json();
  console.log('3. My Assignments:', assignsData.length, 'records');

  if (assignsData.length > 0) {
    const firstAssign = assignsData[0];
    console.log('   First assignment code:', firstAssign.assignment_code);
    
    // Switch Preference
    const patchPref = await fetch('http://localhost:3000/api/platform/preferences', {
      method: 'PATCH',
      headers: { 'Authorization': `Bearer ${token}`, 'Content-Type': 'application/json' },
      body: JSON.stringify({
        service_id: 'church_think',
        preference_key: 'last_context',
        preference_value: { assignment_id: firstAssign.id }
      })
    });
    const patchRes = await patchPref.json();
    console.log('4. PATCH Preference Result:', patchRes);
    
    // Get Preference
    const getPref = await fetch('http://localhost:3000/api/platform/preferences/church_think/last_context', {
      headers: { 'Authorization': `Bearer ${token}` }
    });
    const getPrefRes = await getPref.json();
    console.log('5. GET Preference Result:', getPrefRes);
  } else {
    console.log('   No assignments found for admin.');
  }

  // Create an assignment
  // Wait, we need an organization. Let's list orgs
  const orgs = await fetch('http://localhost:3000/api/church/admin/committees', {
    headers: { 'Authorization': `Bearer ${token}` }
  });
  const orgData = await orgs.json();
  
  const pos = await fetch('http://localhost:3000/api/church/positions', {
    headers: { 'Authorization': `Bearer ${token}` }
  });
  const posData = await pos.json();

  if (orgData.length > 0 && posData.length > 0) {
    const createAssign = await fetch(`http://localhost:3000/api/church/assignments/users/${adminData.user.id}`, {
      method: 'POST',
      headers: { 'Authorization': `Bearer ${token}`, 'Content-Type': 'application/json' },
      body: JSON.stringify({
        committee_id: orgData[0].department_id,
        position_id: posData[0].position_id,
        is_primary: false
      })
    });
    const res = await createAssign.json();
    console.log('6. CREATE Assignment with sequence test:', res);
  }
}

testPlatformAssignments();

File: scratch/test_assignments_platform.py
import urllib.request
import urllib.parse
import json

def fetch(url, method='GET', headers=None, data=None):
    if headers is None: headers = {}
    if data is not None:
        data = json.dumps(data).encode('utf-8')
        headers['Content-Type'] = 'application/json'
    req = urllib.request.Request(url, data=data, method=method, headers=headers)
    try:
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode('utf-8'))
    except Exception as e:
        print(f"Error fetching {url}: {e}")
        return None

# 1. Login
login_data = fetch('http://localhost:3000/api/auth/login', 'POST', data={'email': 'admin@boozathink.com', 'password': 'admin'})
if not login_data or 'token' not in login_data:
    print('Login failed.')
    exit(1)

token = login_data['token']
user_id = login_data['user']['id']
auth_headers = {'Authorization': f'Bearer {token}'}
print('1. Logged in as Admin')

# 2. Profile
profile = fetch('http://localhost:3000/api/church/profile', headers=auth_headers)
print('2. Profile:', profile)

# 3. Assignments
assigns = fetch('http://localhost:3000/api/church/assignments/me', headers=auth_headers)
print(f'3. My Assignments: {len(assigns) if assigns else 0} records')

if assigns and len(assigns) > 0:
    first_assign = assigns[0]
    print(f'   First assignment ID: {first_assign.get("id")}')
    print(f'   First assignment code: {first_assign.get("assignment_code")}')
    
    # 4. PATCH Preference
    patch_data = {
        'service_id': 'church_think',
        'preference_key': 'last_context',
        'preference_value': {'assignment_id': first_assign['id']}
    }
    patch_res = fetch('http://localhost:3000/api/platform/preferences', 'PATCH', headers=auth_headers, data=patch_data)
    print('4. PATCH Preference Result:', patch_res)
    
    # 5. GET Preference
    get_pref = fetch('http://localhost:3000/api/platform/preferences/church_think/last_context', headers=auth_headers)
    print('5. GET Preference Result:', get_pref)

# 6. Create Assignment with Sequence Test
orgs = fetch('http://localhost:3000/api/church/admin/committees', headers=auth_headers)
pos = fetch('http://localhost:3000/api/church/positions', headers=auth_headers)

if orgs and pos and len(orgs) > 0 and len(pos) > 0:
    create_data = {
        'committee_id': orgs[0]['department_id'],
        'position_id': pos[0]['position_id'],
        'is_primary': False
    }
    create_res = fetch(f'http://localhost:3000/api/church/assignments/users/{user_id}', 'POST', headers=auth_headers, data=create_data)
    print('6. CREATE Assignment Result (Sequence test):', create_res)
